solutions/kayenta/ci/scripts/automated-canary.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+set -x
+set -e
+
+## Setup ##
+gcloud config set project $GOOGLE_CLOUD_PROJECT
+gcloud config set compute/zone us-central1-f
+gcloud container clusters get-credentials kayenta-tutorial
+
+curl --fail -LO https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
+chmod +x jq-linux64
+mv jq-linux64 /usr/local/bin/jq
+apt-get -y install bc
+
+## Configuring Kayenta ##
+export HALYARD_POD=$(kubectl -n spinnaker get pods -l \
+    stack=halyard,app=spin \
+    -o=jsonpath='{.items[0].metadata.name}')
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google enable"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google account add kayenta-tutorial --project $GOOGLE_CLOUD_PROJECT"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google edit --stackdriver-enabled=true"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal deploy apply"
+
+set +x
+timeout=0
+while [ "$timeout" -lt 600 -a $(kubectl -n spinnaker get pods | grep "kayenta" | wc -l) -ne 1 ]; do
+  echo "[${timeout}s] Spinnaker not yet ready, waiting 15s more."
+  sleep 15
+  timeout=$((timeout+15))
+done
+if [ "$timeout" -ge 600 ]; then
+  echo "Timeout configuring Kayenta"
+  exit 1
+fi
+echo "Spinnaker ready!"
+set -x
+
+## Enabling canary on sampleapp ##
+DECK_POD=$(kubectl -n spinnaker get pods -l \
+    cluster=spin-deck,app=spin \
+    -o=jsonpath='{.items[0].metadata.name}')
+kubectl -n spinnaker port-forward $DECK_POD 8080:9000 \
+    >/dev/null &
+sleep 5
+
+curl -XPOST -d '{"job":[{"type":"updateApplication","application":{"name":"sampleapp","dataSources":{"enabled":["canaryConfigs"],"disabled":[]}},"user":"[anonymous]"}],"application":"sampleapp","description":"Update Application: sampleapp"}' \
+    --fail -sS -H "Content-Type: application/json" -H "Accept: */*" \
+    http://localhost:8080/gate/applications/sampleapp/tasks >/dev/null
```

Comment:
Any reason not to use the `spin` cli for the application & pipeline tasks?

Reply:
If this is only internal it doesn't really matter -- if we plan to make this accessible or share this as part of the solution we should add a TODO to use the recommended tooling.

Reply:
It's only internal. But I used spin where I could, which is almost nowhere:
* This particular line is enabling the canary option for the `sampleapp` app. Not possible with spin.
* I need to pass a parameter when launching a pipeline, spin doesn't support that yet.
* 2 of the pipelines I create have dependencies on other things that I need to dynamically resolve. AFAIK, spin doesn't support that yet.

Reply:
@jtk54 ^

Reply:
@MrTrustor What are the 'other things' that need dynamically resolved? Created tracking bugs for the first two things, thanks for the feedback.